diff --git a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-csi2.c b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-csi2.c
index 486fe810c5c8..1ebf918979d9 100644
--- a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-csi2.c
+++ b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-csi2.c
@@ -206,6 +206,8 @@ static const struct rzg2l_csi2_format rzg2l_csi2_formats[] = {
 	  .rate = RATE_RANGES(80, 4256)},
 	{ .code = MEDIA_BUS_FMT_RGB888_1X24,	.bpp = 24,
 	  .rate = RATE_RANGES(80, 6000)},
+	{ .code = MEDIA_BUS_FMT_BGR888_1X24,	.bpp = 24,
+	  .rate = RATE_RANGES(80, 6000)},
 	{ .code = MEDIA_BUS_FMT_SBGGR8_1X8,	.bpp = 8,
 	  .rate = RATE_RANGES(80, 2128)},
 	{ .code = MEDIA_BUS_FMT_SGBRG8_1X8,	.bpp = 8,
@@ -214,6 +216,8 @@ static const struct rzg2l_csi2_format rzg2l_csi2_formats[] = {
 	  .rate = RATE_RANGES(80, 2128)},
 	{ .code = MEDIA_BUS_FMT_SRGGB8_1X8,	.bpp = 8,
 	  .rate = RATE_RANGES(80, 2128)},
+	{ .code = MEDIA_BUS_FMT_YUYV8_2X8,	.bpp = 16,
+	  .rate = RATE_RANGES(80, 4256)},
 	{ .code = MEDIA_BUS_FMT_YUYV10_2X10,	.bpp = 20,
 	  .rate = RATE_RANGES(80, 5238)},
 	{ .code = MEDIA_BUS_FMT_SRGGB10_1X10,	.bpp = 10,
-- 
2.34.1

