diff --git a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-ip.c b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-ip.c
index 727f43a7bedf..19839d06ef87 100644
--- a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-ip.c
+++ b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-ip.c
@@ -18,6 +18,15 @@ static const struct rzg2l_cru_ip_format rzg2l_cru_ip_formats[] = {
 		.icndmr = ICnDMR_YCMODE_UYVY,
 		.fmt_types = V4L2_PIXEL_ENC_YUV,
 	},
+        {
+                .code = MEDIA_BUS_FMT_YUYV8_2X8,
+                .datatype = ICnMC_INF_YUV8_422,
+                .format = V4L2_PIX_FMT_YUYV,
+                .bpp = 2,
+                .icndmr = ICnDMR_YCMODE_YUYV,
+                .fmt_types = V4L2_PIXEL_ENC_YUV,
+        },
+	
 	{
 		.code = MEDIA_BUS_FMT_UYVY8_1X16,
 		.datatype = ICnMC_INF_YUV8_422,
@@ -66,6 +75,14 @@ static const struct rzg2l_cru_ip_format rzg2l_cru_ip_formats[] = {
 		.icndmr = ICnDMR_RGBMODE_RGB24,
 		.fmt_types = V4L2_PIXEL_ENC_RGB,
 	},
+        {
+                .code = MEDIA_BUS_FMT_BGR888_1X24,
+                .datatype = ICnMC_INF_RGB888,
+                .format = V4L2_PIX_FMT_BGR24,
+                .bpp = 3,
+                .icndmr = ICnDMR_RGBMODE_RGB24,
+                .fmt_types = V4L2_PIXEL_ENC_RGB,
+        },
 	{
 		.code = MEDIA_BUS_FMT_RGB444_1X12,
 		.datatype = ICnMC_INF_RGB444,
diff --git a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-video.c b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-video.c
index 6f0e58a76a6a..3ba6005c4212 100644
--- a/drivers/media/platform/renesas/rzg2l-cru/rzg2l-video.c
+++ b/drivers/media/platform/renesas/rzg2l-cru/rzg2l-video.c
@@ -1384,6 +1384,7 @@ static int rzg2l_cru_video_link_validate(struct media_link *link)
 	} else {
 		switch (fmt.format.code) {
 		case MEDIA_BUS_FMT_UYVY8_2X8:
+		case MEDIA_BUS_FMT_YUYV8_2X8:
 		case MEDIA_BUS_FMT_UYVY10_2X10:
 		case MEDIA_BUS_FMT_UYVY8_1X16:
 		case MEDIA_BUS_FMT_YUYV8_1X16:
@@ -1392,6 +1393,7 @@ static int rzg2l_cru_video_link_validate(struct media_link *link)
 		case MEDIA_BUS_FMT_RGB565_2X8_LE:
 		case MEDIA_BUS_FMT_RGB666_1X18:
 		case MEDIA_BUS_FMT_RGB888_1X24:
+		case MEDIA_BUS_FMT_BGR888_1X24:	
 		case MEDIA_BUS_FMT_SBGGR8_1X8:
 		case MEDIA_BUS_FMT_SGBRG8_1X8:
 		case MEDIA_BUS_FMT_SGRBG8_1X8:
-- 
2.34.1

